product_route_profile/models/product_template.py

```diff
@@ -45,19 +49,23 @@ def _search_route_ids(self, operator, value):
             ("route_profile_id.route_ids", operator, value),
         ]
 
+    def _get_or_create_profile(self, profiles, routes):
+        self.ensure_one()
+        if not routes:
+            return self.env["route.profile"]
+        for profile in profiles:
+            if routes == profile.route_ids:
+                return profile
+        vals = self._prepare_profile(routes)
+        return self.env["route.profile"].create(vals)
+
     def _inverse_route_ids(self):
         profiles = self.env["route.profile"].search([])
```

Comment:
This makes me think: to limit the search, this could be great:

```suggestion
        profiles = self.env["route.profile"].search([("route_ids", "in", self.mapped("route_ids").ids])
```

But never tested such query